mono/metadata/object.c

```diff
@@ -436,7 +436,10 @@ mono_runtime_class_init_full (MonoVTable *vtable, MonoError *error)
 		MonoException *exc = NULL;
 
 		mono_threads_begin_abort_protected_block ();
-		mono_runtime_try_invoke (method, NULL, NULL, (MonoObject**) &exc, error);
+		/* Its possible that we were already being aborted before calling begin_abort_protected_block () */
+		exc = mono_thread_interruption_checkpoint ();
```

Comment:
This will fail loading the type if there's a pending exception, which is exactly what the abort protected block is  meant to avoid.

We try to pic a pending TAE at the end by calling mono_thread_try_resume_interruption.